### PR TITLE
feat: switch to `abstract-level`

### LINKED ISSUE #42: switch to `abstract-level` from `abstract-leveldown`
It looks like the level community is deprecating [`abstract-leveldown`](https://github.com/Level/abstract-leveldown) in favor of [`abstract-level`](https://github.com/Level/abstract-level). They published some [nice upgrade guides](https://github.com/Level/community#how-do-i-upgrade-to-abstract-level) that look straight forward enough.

### PR DESCRIPTION
**Pull request by mikaelkaron**
closes #42

### CODE REVIEW

**Comment by mikaelkaron** [comment]
I took a quick stab at it. Have to admit I have _no_ idea what I'm actually doing, but all the tests are passing...

File: package.json [hunk 1]
```diff
@@ -5,19 +5,18 @@
...
+    "classic-level": "^1.2.0",
+    "level-read-stream": "^1.1.0",
+    "memory-level": "^1.0.0",
     "traverse": "^0.6.6"
   },
   "devDependencies": {
     "assert": "^2.0.0",
     "buffer": "^6.0.3",
     "diacritic": "^0.0.2",
-    "level-out": "^1.0.1",
```
Comment (mikaelkaron) [inline package.json:19]: I never saw this used in the code so I just removed it

File: webpack.config.js [hunk 2]
```diff
@@ -24,7 +24,8 @@ const config = {
       "fs": false,
       "path": require.resolve("path-browserify"),
       "stream": require.resolve("stream-browserify"),
-      'util': false
+      'util': false,
+      "os": require.resolve("os-browserify")
```
Comment (mikaelkaron) [inline webpack.config.js:28]: I've never messed around with webpack before, but this is close to what the CLI suggested when it threw an error at me (I think the actual require was `os-browserify/browser` but I followed what was already in the file)

**Comment by fergiemcdowall** [comment]
Thanks for the PR @mikaelkaron . I am currently away from home without my laptop until Saturday, but I will take a proper look at it then.

It looks like the tests are failing on a small linting (`standard`) error.

**Comment by mikaelkaron** [comment]
Yeah, see that now. So the error thrown is [`A constructor name should not start with a lowercase letter`](https://github.com/fergiemcdowall/fergies-inverted-index/runs/7438905405?check_suite_focus=true#step:6:829) caused by the fact that you have to `new` the `abstract-level` implementations and the old code was called `ops.db`.

I can solve this by:
- renaming `ops.db` to `ops.DB` (or anything starting with upper case)
- temporarily storing `ops.db` in another `const DB = ops.db` before newing

I'll make a small commit using the second option to change the "public" api as little as possible

**Comment by fergiemcdowall** [comment]
Tests look good 👍I will take a look at the rest ASAP

File: webpack.config.js (hunk 2 above)
Comment (fergiemcdowall) [reply webpack.config.js:28]: If it works- great. Seems reasonable. (does anybody truly understand Webpack? 😅😅)

File: package.json (hunk 1 above)
Comment (fergiemcdowall) [reply package.json:19]: Yeah- I personally like to use `level-out`as a dev tool to get a quick look at the index, but you are right- it shouldn't be an actual dev dependency

**Comment by mikaelkaron** [comment]
since this will (probably) result in a breaking change, I'd suggest a small API refactoring as well.

Previously it might have made sense to send the "creator` function as the DB, but since [the new `abstract-level` API forwards the constructor parameters to the `open` method](https://github.com/Level/abstract-level/blob/main/UPGRADING.md#14-new-deferred-open) it makes more sense (imho) to provide an already instantiated `db` instead of the constructor.

**Comment by fergiemcdowall** [comment]
> it makes more sense (imho) to provide an already instantiated db instead of the constructor.

Could you provide an example of how this would/should work?

**Comment by mikaelkaron** [comment]
> > it makes more sense (imho) to provide an already instantiated db instead of the constructor.
> 
> Could you provide an example of how this would/should work?

stared a PR in https://github.com/fergiemcdowall/fergies-inverted-index/pull/45

**Pull request merged**